fitbenchmarking/fitbenchmarking
# Change defaults to pip installables only

#### Description of Work

Fixes #343 

Remove gsl from setup.py
Update defaults to pip installable modules only
Edit docs to state that some modules are not installed by default

#### Testing Instructions

1. Check installs work

Function: Does the change do what it's supposed to?
Tests: Does it pass? Is there adequate coverage for new code?
Style: Is the coding style consistent? Is anything overly confusing?
Documentation: Is there a suitable change to documentation for this change?
